pkg/alibabacloud/eni/node_test.go

```diff
@@ -122,33 +130,35 @@ func (e *ENISuite) TestCandidateAndEmtpyInterfaces(c *check.C) {
 	n := &Node{}
 	n.k8sObj = cn
 	// Primary ENI excluded, max allocatable = 3 ( 1 (ENI) * 3 (IPv4/ENI) )
-	c.Assert(n.GetMaximumAllocatableIPv4(), check.Equals, 3)
+	require.Equal(t, n.GetMaximumAllocatableIPv4(), 3)
 
 	// Wait for IPs to become available
-	c.Assert(testutils.WaitUntil(func() bool { return reachedAddressesNeeded(mngr, "node3", 0) }, 5*time.Second), check.IsNil)
+	require.NoError(t, testutils.WaitUntil(func() bool { return reachedAddressesNeeded(mngr, "node3", 0) }, 5*time.Second))
 
 	node3 := mngr.Get("node3")
 	a, err := node3.Ops().PrepareIPAllocation(log)
-	c.Assert(err, check.IsNil)
+	require.NoError(t, err)
 	// 1 ENI attached, 1/3 IPs allocated, 0 empty slots left
-	c.Assert(a.IPv4.InterfaceCandidates, check.Equals, 1)
-	c.Assert(a.EmptyInterfaceSlots, check.Equals, 0)
-	c.Assert(node3.Stats().IPv4.AvailableIPs, check.Equals, 1)
+	require.Equal(t, a.IPv4.InterfaceCandidates, 1)
+	require.Equal(t, a.EmptyInterfaceSlots, 0)
+	require.Equal(t, 1, node3.Stats().IPv4.AvailableIPs)
 }
 
-func (e *ENISuite) TestPrepareIPAllocation(c *check.C) {
+func TestPrepareIPAllocation(t *testing.T) {
+	setup(t)
+
 	alibabaAPI.UpdateENIs(primaryENIs)
 	instances.Resync(context.TODO())
 
 	mngr, err := ipam.NewNodeManager(instances, k8sapi, metricsapi, 10, false, false)
-	c.Assert(err, check.IsNil)
-	c.Assert(mngr, check.Not(check.IsNil))
+	require.NoError(t, err)
+	require.NotNil(t, mngr)
 	mngr.SetInstancesAPIReadiness(false) // to avoid the manager background jobs starting and racing us.
 
 	mngr.Upsert(newCiliumNode("node1", "i-1", "ecs.g7ne.large", "cn-hangzhou-i", "vpc-1"))
 	a, err := mngr.Get("node1").Ops().PrepareIPAllocation(log)
-	c.Assert(err, check.IsNil)
-	c.Assert(a.EmptyInterfaceSlots+a.IPv4.InterfaceCandidates, check.Equals, 2, check.Commentf("empty: %v, candidates: %v", a.EmptyInterfaceSlots, a.IPv4.InterfaceCandidates))
+	require.NoError(t, err)
+	require.Equal(t, 0, a.IPv4.InterfaceCandidates, fmt.Sprintf("empty: %v, candidates: %v", a.EmptyInterfaceSlots, a.IPv4.InterfaceCandidates))
```

Comment:
nit: you can use `assert.Equalf`
```suggestion
	require.Equalf(t, 0, a.IPv4.InterfaceCandidates, "empty: %v, candidates: %v", a.EmptyInterfaceSlots, a.IPv4.InterfaceCandidates)
```

Also, did you mean to change the assertion? It looks like before it was checking that `a.EmptyInterfaceSlots+a.IPv4.InterfaceCandidates == 2` but now it checks `a.IPv4.InterfaceCandidates == 0`?

Reply:
good catch :+1: